pydantic/mypy.py

```diff
@@ -857,9 +855,6 @@ def add_initializer(
                 use_alias=use_alias,
                 is_settings=is_settings,
             )
-            if is_root_model:
-                # convert root argument to positional argument
-                args[0].kind = ARG_POS if args[0].kind == ARG_NAMED else ARG_OPT
```

Comment:
@Viicos,

If we add this back, do we get no errors for `mypy 1.0.1` as well as all of the other `mypy` versions currently tested?

If so, I'd prefer to do that + add a note about the special case for 1.0.1.

Reply:
Or if necessary, wrap this check in a block that checks the `mypy` version